code/modules/power/supermatter/supermatter.dm

```diff
@@ -698,6 +700,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		removed.gases[/datum/gas/plasma][MOLES] += max((device_energy * dynamic_heat_modifier) / PLASMA_RELEASE_MODIFIER, 0)
 		//Varies based on power, gas content, and heat
 		removed.gases[/datum/gas/oxygen][MOLES] += max(((device_energy + removed.temperature * dynamic_heat_modifier) - T0C) / OXYGEN_RELEASE_MODIFIER, 0)
+		//Aboce certain power thresholds the supermatter crystal can electrolyze water vapor.
+		if(power > POWER_PENALTY_THRESHOLD)
+			var/h2o_moles = removed.gases[/datum/gas/water_vapor][MOLES]
+			var/electrolysis_moles = max(h2o_moles * (power - POWER_PENALTY_THRESHOLD)/(power + WATER_ELECTROLYSIS_MODIFIER), 0)
+			if(electrolysis_moles)
+				removed.gases[/datum/gas/water_vapor][MOLES] -= electrolysis_moles
```

Comment:
use `removed.gases[/datum/gas/water_vapor][MOLES] -= min(h2o_moles, electrolysis_moles)` this way you'd be sure to not go under 0 (yes it should never go from how you set the electrolysis_moles math, but unless this is costly, better safe than sorry) or at least is what i would do..

Reply:
I'll clamp it when I set up electrolysis moles.

Reply:
👍